wren-sqlglot-converter/src/main/java/io/wren/sqlglot/glot/SQLGlot.java

```diff
@@ -0,0 +1,107 @@
+import io.airlift.http.client.HttpClientConfig;
+import io.airlift.http.client.Request;
+import io.airlift.http.client.StringResponseHandler;
+import io.airlift.http.client.jetty.JettyHttpClient;
+import io.airlift.json.JsonCodec;
+import io.airlift.units.Duration;
+import io.wren.sqlglot.dto.TranspileDTO;
+
+import javax.ws.rs.WebApplicationException;
+import javax.ws.rs.core.UriBuilder;
+
+import java.io.Closeable;
+import java.io.IOException;
+import java.net.URI;
+import java.util.Optional;
+
+import static io.airlift.http.client.JsonBodyGenerator.jsonBodyGenerator;
+import static io.airlift.http.client.Request.Builder.preparePost;
+import static io.airlift.http.client.StringResponseHandler.createStringResponseHandler;
+import static io.airlift.json.JsonCodec.jsonCodec;
+import static java.util.concurrent.TimeUnit.SECONDS;
+import static javax.ws.rs.core.HttpHeaders.CONTENT_TYPE;
+import static javax.ws.rs.core.MediaType.APPLICATION_JSON;
+
+public class SQLGlot
+        implements Closeable
+{
+    public enum Dialect
+    {
+        BIGQUERY("bigquery"),
+        DUCKDB("duckdb"),
+        POSTGRES("postgres"),
+        SNOWFLAKE("snowflake"),
+        TRINO("trino");
+
+        private final String dialect;
+
+        Dialect(String dialect)
+        {
+            this.dialect = dialect;
+        }
+
+        public String getDialect()
+        {
+            return dialect;
+        }
+    }
+
+    public static final URI BASE_URL = URI.create("http://0.0.0.0:" + getPort());
+
+    private static final JsonCodec<TranspileDTO> TRANSPILE_DTO_JSON_CODEC = jsonCodec(TranspileDTO.class);
+
+    private final HttpClient client;
+
+    public SQLGlot()
+    {
+        this.client = new JettyHttpClient(new HttpClientConfig().setIdleTimeout(new Duration(20, SECONDS)));
+    }
+
+    @Override
+    public void close()
+    {
+        client.close();
+    }
+
+    public String transpile(String sql, Dialect read, Dialect write)
+            throws IOException
+    {
+        Request request = preparePost()
+                .setUri(UriBuilder.fromUri(BASE_URL).path("sqlglot").path("transpile").build())
+                .setHeader(CONTENT_TYPE, APPLICATION_JSON)
+                .setBodyGenerator(jsonBodyGenerator(TRANSPILE_DTO_JSON_CODEC, new TranspileDTO(sql, read.getDialect(), write.getDialect())))
+                .build();
+        StringResponseHandler.StringResponse response = client.execute(request, createStringResponseHandler());
+        if (response.getStatusCode() != 200) {
+            throwWebApplicationException(response);
+        }
+        return TRANSPILE_DTO_JSON_CODEC.fromJson(response.getBody()).getSql();
+    }
+
+    private static void throwWebApplicationException(StringResponseHandler.StringResponse response)
+    {
+        throw new WebApplicationException(response.getBody());
+    }
+
+    public static String getPort()
+    {
+        return Optional.ofNullable(System.getenv("SQLGLOT_PORT")).orElse("8000");
```

Comment:
Instead of `env`, I prefer to use `Config`. It would be good to trace those exist configuration.